crates/nu-cli/src/commands/history_session.rs

```diff
@@ -18,25 +23,55 @@ impl Command for HistorySession {
 
     fn signature(&self) -> nu_protocol::Signature {
         Signature::build("history session")
+            .named("set", SyntaxShape::Int, "Set the session_id to", Some('s'))
             .category(Category::Misc)
-            .input_output_types(vec![(Type::Nothing, Type::Int)])
+            .input_output_types(vec![(Type::Nothing, Type::Int), (Type::Int, Type::Nothing)])
     }
 
     fn examples(&self) -> Vec<Example> {
-        vec![Example {
-            example: "history session",
-            description: "Get current history session",
-            result: None,
-        }]
+        vec![
+            Example {
+                example: "history session",
+                description: "Get current history session",
+                result: None,
+            },
+            Example {
+                example: "history -l | where session_id == (history session) | last 5",
+                description: "Gets the last 5 history entries of the current session",
+                result: None,
+            },
+            Example {
+                example: "history session --set (history session)",
+                description: "Sets the history session to a different history session (example
+            sets it to the same history session)",
+                result: None,
+            },
+        ]
     }
 
     fn run(
         &self,
         engine_state: &EngineState,
-        _stack: &mut Stack,
+        stack: &mut Stack,
         call: &Call,
         _input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
-        Ok(Value::int(engine_state.history_session_id, call.head).into_pipeline_data())
+        let set_session: Option<Value> = call.get_flag(engine_state, stack, "set")?;
+        if let Some(set_session) = set_session {
+            let set_session_id = set_session.as_i64()?;
+            #[allow(mutable_transmutes)]
+            let engine_state =
+                unsafe { std::mem::transmute::<&EngineState, &mut EngineState>(engine_state) };
```

Comment:
**Stop!** This code is clearly unsound: This breaks aliasing rules and is [explicitly called out as UB](https://doc.rust-lang.org/nomicon/transmutes.html). 
Interior mutability has to either use the [`UnsafeCell`](https://doc.rust-lang.org/std/cell/struct.UnsafeCell.html) intrinsic or is ok for atomic read/writes as found with the types in [`std::sync::atomic`](https://doc.rust-lang.org/std/sync/atomic/index.html). 

Before trying to do something like this, we can always fall back to `Arc<Mutex<T>>` and friends with explicit APIs to do safe mutation given a shared reference.